rust/worker/src/execution/orchestration/compact.rs

```diff
@@ -765,17 +617,39 @@ impl Handler<TaskResult<MaterializeLogsResult, MaterializeLogOperatorError>>
             {
                 // There is nothing to flush, proceed to register
                 self.register(self.pulled_log_offset.unwrap(), ctx).await;
+                return;
             }
         } else {
-            self.dispatch_apply_log_to_segment_writer_tasks(
-                materialized_result,
-                ctx.receiver(),
-                ctx,
-            )
-            .await;
+            self.materialized_results.push(materialized_result);
         }
 
         self.num_uncompleted_materialization_tasks -= 1;
+
+        if self.num_uncompleted_materialization_tasks == 0 {
```

Comment:
technically this slightly reduces parallelism since it adds a sync point and waits for all partitions to finish materializing before starting to apply the log

I couldn't think of a clean way to retain parallelism while only getting the writers once.